src/metabase/api/search.clj

```diff
@@ -401,42 +408,50 @@
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
 (mu/defn ^:private search-context
-  [{:keys [archived-string
+  [{:keys [archived
            created-by
            limit
            models
            offset
            search-string
-           table-db-id]} :- [:map {:closed true}
-                             [:search-string                    [:maybe ms/NonBlankString]]
-                             [:models                           [:maybe [:set SearchableModel]]]
-                             [:archived-string {:optional true} [:maybe ms/BooleanString]]
-                             [:created-by      {:optional true} [:maybe ms/PositiveInt]]
-                             [:limit           {:optional true} [:maybe ms/Int]]
-                             [:offset          {:optional true} [:maybe ms/Int]]
-                             [:table-db-id     {:optional true} [:maybe ms/PositiveInt]]]]
+           table-db-id
+           verified]}      :- [:map {:closed true}
+                               [:search-string                    [:maybe ms/NonBlankString]]
+                               [:models                           [:maybe [:set SearchableModel]]]
+                               [:archived        {:optional true} [:maybe :boolean]]
+                               [:created-by      {:optional true} [:maybe ms/PositiveInt]]
+                               [:limit           {:optional true} [:maybe ms/Int]]
+                               [:offset          {:optional true} [:maybe ms/Int]]
+                               [:table-db-id     {:optional true} [:maybe ms/PositiveInt]]
+                               [:verified        {:optional true} [:maybe true?]]]]
   (let [models (if (string? models) [models] models)
         ctx    (cond-> {:search-string      search-string
                         :current-user-perms @api/*current-user-permissions-set*
-                        :archived?          (Boolean/parseBoolean archived-string)
+                        :archived?          (boolean archived)
                         :models             models}
                  (some? created-by)  (assoc :created-by created-by)
                  (some? table-db-id) (assoc :table-db-id table-db-id)
                  (some? limit)       (assoc :limit-int limit)
-                 (some? offset)      (assoc :offset-int offset))]
+                 (some? offset)      (assoc :offset-int offset)
+                 (some? verified)    (assoc :verified (or (premium-features/assert-has-any-features
```

Comment:
Having the assert in an `or` expression hurts readabiility from my perspective. Can you make the assert statement somewhere else?

Reply:
how about at the top level of `search-context` function? https://github.com/metabase/metabase/blob/caa17f36206a1cc74a5e7d52022bb2e9fb5ef28e/src/metabase/api/search.clj#L427

Reply:
Yes, I think that's tidier